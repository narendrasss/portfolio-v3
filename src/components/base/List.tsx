import React from 'react';

export interface ListProps<T> {
  items: T[];
  children(item: T): JSX.Element;
}

const List = <T extends {}>(props: ListProps<T>) => {
  const { items, children } = props;
  return <>{items.map(item => children(item))}</>;
};

export default List;
