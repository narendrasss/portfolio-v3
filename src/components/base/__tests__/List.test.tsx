import React from 'react';
import { render } from 'react-testing-library';
import List from '../List';

describe('List', () => {
  const testArray = ['hello', 'welcome'];
  const renderFn = (str: string) => <p key={str}>{str}</p>;

  it('renders without crashing', () => {
    const { asFragment } = render(<List items={testArray}>{renderFn}</List>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the correct elements', () => {
    const { queryByText } = render(<List items={testArray}>{renderFn}</List>);
    testArray.forEach(str => {
      expect(queryByText(str)).not.toBeNull();
    });
  });
});
