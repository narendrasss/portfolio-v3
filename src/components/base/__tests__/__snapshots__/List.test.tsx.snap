// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders without crashing 1`] = `
<DocumentFragment>
  <p>
    hello
  </p>
  <p>
    welcome
  </p>
</DocumentFragment>
`;
